src/QrSigner.js

```diff
@@ -101,39 +91,17 @@ export default class QrSigner extends Component {
       );
     }
 
-    let { account, rlp, data } = this.props;
-    let value;
+    const { network, payload } = this.props;
+    const value = encode(network, payload);
 
-    if (!account || !(rlp || data)) {
-      console.error('Missing `account`, `rlp` or `data` prop on QrSigner!');
-
-      return null;
-    }
-
-    account = remove0x(account);
-
-    if (rlp) {
-      rlp = remove0x(rlp);
-
-      value = {
-        action: 'signTransaction',
-        data: { account, rlp }
-      };
-    } else {
-      data = remove0x(data);
-
-      value = {
-        action: 'signData',
-        data: { account, data }
-      };
+    if (value.error) {
+      onError && onError(value.error);
+      return <div style={style}>Error in encoding: {value.error}</div>;
```

Comment:
What should we do in case the user inputs a wrong network/payload combo? Right now I'm just showing this ugly div, and also added an `onError` prop

Reply:
I think that's reasonable, the burden of correctness is on the API consumer (!== end user) in this case.